contracts/messaging/gateway/FuelERC20Gateway.sol

```diff
@@ -125,12 +169,35 @@ contract FuelERC20Gateway is
         //reduce deposit balance and transfer tokens (math will underflow if amount is larger than allowed)
         _deposits[tokenId][fuelTokenId] = _deposits[tokenId][fuelTokenId] - amount;
         IERC20Upgradeable(tokenId).safeTransfer(to, amount);
+
+        //emit event for successful token withdraw
+        emit Withdrawal(bytes32(uint256(uint160(to))), tokenId, fuelTokenId, amount);
     }
 
     ////////////////////////
     // Internal Functions //
     ////////////////////////
 
+    /// @notice Deposits the given tokens to an account or contract on Fuel
+    /// @param tokenId ID of the token being transferred to Fuel
+    /// @param fuelTokenId ID of the token on Fuel that represent the deposited tokens
+    /// @param amount Amount of tokens to deposit
+    /// @param messageData The data of the message to send for deposit
+    /// @dev Made payable to reduce gas costs
```

Comment:
Is this comment out of date? A private function can't be payable can it?

Reply:
good catch